metaci/api/views/testmethod_perf.py

```diff
@@ -0,0 +1,284 @@
+    )
+
+    group_by = django_filters.rest_framework.MultipleChoiceFilter(
+        label="Split On (multi-select okay)", choices=group_by_choices, method=dummy
+    )
+
+    metrics = {
+        "average_duration": Avg("duration"),
+        "slowest_runs": Min("duration"),  # fixme: use p05
+        "fastest_runs": Max("duration"),  # fixme: use p95
+        "average_cpu_usage": Avg("test_cpu_time_used"),
+        "low_cpu_usage": Min("test_cpu_time_used"),
+        "high_cpu_usage": Max("test_cpu_time_used"),
+        "count": Count("id"),
+        "failures": Count("id", filter=Q(outcome="Fail")),
+        "assertion_failures": Count(
+            "id", filter=Q(message__startswith="System.AssertException")
+        ),
+        "DML_failures": Count(
+            "id", filter=Q(message__startswith="System.DmlException")
+        ),
+        "Other_failures": Count(
+            "id",
+            filter=~Q(message__startswith="System.DmlException")
+            & ~Q(message__startswith="System.AssertException"),
+        ),
+        "success_percentage": Cast(Count("id", filter=Q(outcome="Pass")), FloatField())
+        / Cast(Count("id"), FloatField()),
+    }
+
+    metric_choices = tuple(zip(metrics.keys(), metrics.keys()))
+    include_fields = django_filters.rest_framework.MultipleChoiceFilter(
+        label="Include (multi-select okay)", choices=metric_choices, method=dummy
+    )
+
+    o = django_filters.rest_framework.OrderingFilter(
+        fields=(
+            metric_choices
+            + group_by_choices
+            + (
+                ("method_name", "method_name"),
+                ("count", "count"),
+                ("failures", "failures"),
+                ("assertion_failures", "assertion_failures"),
+                ("DML_failures", "DML_failures"),
+                ("Other_failures", "Other_failures"),
+                ("success_percentage", "success_percentage"),
+            )
+        )
+    )
+
+
+BUILD_FLOWS_LIMIT = 100
+
+
+class TestMethodPerfListView(generics.ListAPIView):
+    """
+    A view for lists of aggregated test metrics.
+
+    Note that the number of build flows covered is limited to **BUILD_FLOWS_LIMIT** for performance reasons. You can
+    change this default with the build_flows_limit parameter.
+    """
+
+    serializer_class = TestMethodPerfSerializer
+    filter_backends = (django_filters.rest_framework.DjangoFilterBackend,)
+    filter_class = TestMethodPerfFilter
+
+    # example URLs:
+    # http://localhost:8000/api/testmethod_perf/?repo=gem&plan=Release%20Test&method_name=testCreateNegative
+    # http://localhost:8000/api/testmethod_perf/?repo=Cumulus&plan=Feature%20Test&o=avg
+    # http://localhost:8000/api/testmethod_perf/?method_name=&repo=&plan=&flow=&recentdate=&daterange_after=&daterange_before=&o=-repo
+
+    def _get_buildflows(self):
+        """Which buildflows do we need to look at? Limit by time, repo, etc.
+           Also limit # for performance reasons"""
```

Comment:
https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings recommends indenting subsequent docstring lines to the same level as the quotes on the first line, and putting the closing quotation marks on their own line.